site/releases/2024-jan-22/highlights.qmd

```diff
@@ -0,0 +1,218 @@
+
+```python
+vm_model = vm.init_model(
+    model,
+    train_ds=vm_train_ds,
+    test_ds=vm_test_ds,
+)
+```
+
+Internally, this example invokes the `predict()` method of the model for the training and test datasets when the model is initialized. This approach can be problematic with large datasets: `init_model` can simply take too long to compute.
+
+You can now avoid this issue by providing a dataset with a column containing pre-computed predictions, similar to the following example. If `init_model` detects this column, it will not generate model predictions at runtime. 
+
+|x1|x2|...|target_column|prediction_column|
+|-|-|-|-|-|
+|0.1|0.2|...|0|0|
+|0.2|0.4|1...|1|1|
+
+Usage example with a prediction column:
+
+```python
+vm.init_dataset(
+     dataset=df,
+     feature_columns=[...],
+     target_column= ...,
+     extra_columns={
+         prediction_column: 'NAME-OF-PREDICTION-COLUMN',
+    },
+)
+```
+
+<!---[SC-2438] Support for adding feature columns parameter to init dataset by @juanmleng in [#279](https://github.com/validmind/developer-framework/pull/279)--->
+#### Improvements to `init dataset`
+
+When initializing a dataset, the new `feature_columns` argument lets you specify a list of feature names for prediction to improve efficiency. Internally, the function filters the dataset to retain only these specified features for prediction-related tasks, leaving the remaining dataset available for other purposes, such as segmentation. 
+
+This improvement replaces the existing behavior of `init_dataset`, which loaded the entire dataset, incorporating all available features for prediction tasks. While this approach worked well, it could impose limitations when generating segmented metrics and proved somewhat inefficient with large datasets containing numerous features, of which only a small subset were relevant for prediction.
+
+Usage example:
+
+```python
+feature_columns = ['CreditScore', 'Age', 'Balance', 'NumOfProducts', 'EstimatedSalary']
+
+vm_train_ds = vm.init_dataset(
+    dataset=train_df,
+    target_column=demo_dataset.target_column,
+    feature_columns=feature_columns
+)
+```
+
+Additionally, two new metrics enable you to report by segment and show the use of `feature_columns`:
+
+- `MissingValuesBySegment`
+- `DescriptiveStatisticsBySegment`
+
+A new notebook illustrates how you can configure these dataset features:
+
+- How to utilize the `feature_columns` parameter when initizalizing `validmind` datasets and model objects
+- How `feature_columns` can be used to report by segment
+
+Try it: [Configuring and Using Dataset Features](../../notebooks/how_to/configure_dataset_features.ipynb)
+
+<!---[SC-2707] Support for multiple sections in run_documentation_tests() by @juanmleng in [#307](https://github.com/validmind/developer-framework/pull/307)--->
+#### Improvements to `run_documentation_tests()`
```

Comment:
description in this highlight is unclear to me. 
- what is the benefit for the user of running multiple sections at a time?
- I don't think we need to specify that this came through a customer feedback (makes it sound like the other improvements are not customer driven)

Reply:
> * what is the benefit for the user of running multiple sections at a time?

When I was working on these release notes, I traced this back to https://app.shortcut.com/validmind/story/2707/run-documentation-tests-should-run-for-multiple-sections, but that's all the info there is. It seem mainly a convenience feature ("...  it would be beneficial ...". 

> * I don't think we need to specify that this came through a customer feedback (makes it sound like the other improvements are not customer driven)

Removed.

Reply:
> what is the benefit for the user of running multiple sections at a time?

The user doesn't need to call the same function twice for two different sections, resulting in code that is less prone to errors: less duplication and ability to use a single config object. Something like this:

## Before

```python
full_suite = vm.run_documentation_tests(
    inputs = {
        ...
    },
    section="section_1",
    config={
        "validmind.tests.data_validation.ClassImbalance": ...
    } 
)

full_suite = vm.run_documentation_tests(
    inputs = {
        ...
    },
    section="section_2",
    config={
        "validmind.tests.data_validation.Duplicates": ...
    } 
)
```

## After

Single function call:

```python
full_suite = vm.run_documentation_tests(
    inputs = {
        ...
    },
    section=["section_1", "section_2"],
    config={
        "validmind.tests.data_validation.ClassImbalance": ...,
        "validmind.tests.data_validation.Duplicates": ...
    } 
)
```

Reply:
Thank you, this is very helpful. I wove that explanation into the text and included the before/after example usage to illustrate.